src/bosh.js

```diff
@@ -632,6 +636,9 @@ Strophe.Bosh.prototype = {
                      this._requests[0].age() > Math.floor(Strophe.SECONDARY_TIMEOUT * this.wait))) {
                     this._restartRequest(0);
                 }
+
+                this._conn.ridChange(Number(req.rid) + 1);
+
```

Comment:
What I don't like about the ``+1`` here is that comes across as dishonest :)

My understanding is that ``ridChange`` should be called whenever the rid has been updated. In this case, the rid doesn't get updated but you pretend that it has, thereby fooling the user of the ``ridChange`` method.

Instead of taking this approach here, what if you rather called ``ridChange`` inside the Strophe.Request constructor? That is the point where we know what the RID of the next request will be, and we know that the RID has been updated at this point.

This means that the RID you store in your app is always the last used RID, which is also more accurate and more honest in my opinion. Since you know it's always the last used RID, you simply increment it yourself when you want to reattach to your BOSH connection.